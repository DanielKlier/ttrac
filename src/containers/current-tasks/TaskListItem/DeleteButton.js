import React from 'react';
import PropTypes from 'prop-types';
import {Button, Glyphicon} from 'react-bootstrap';

const DeleteButton = props => (
    <Button onClick={() =>  typeof props.onClick === 'function' && props.onClick()}>
        <Glyphicon glyph="trash"/>
    </Button>
);

DeleteButton.propTypes = {
    onClick: PropTypes.func
};

export default DeleteButton;
