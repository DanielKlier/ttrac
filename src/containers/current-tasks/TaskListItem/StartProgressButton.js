import React from 'react';
import PropTypes from 'prop-types';
import {Button, Glyphicon} from 'react-bootstrap';
import {noop} from 'lodash';

const StartProgressButton = props => (
    <Button onClick={() => props.onClick()}>
        <Glyphicon glyph="time"/>
    </Button>
);

StartProgressButton.propTypes = {
    onClick: PropTypes.func
};

StartProgressButton.defaultProps = {
    onClick: noop
};

export default StartProgressButton;
