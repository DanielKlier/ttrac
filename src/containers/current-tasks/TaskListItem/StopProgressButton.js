import React from 'react';
import PropTypes from 'prop-types';
import {Button, Glyphicon} from 'react-bootstrap';

const StopProgressButton = props => (
    <Button onClick={() => props.onClick()}>
        <Glyphicon glyph="pause"/>
    </Button>
);

StopProgressButton.propTypes = {
    onClick: PropTypes.func
};

export default StopProgressButton;
