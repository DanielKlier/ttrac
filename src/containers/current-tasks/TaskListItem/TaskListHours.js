import * as React from 'react';
import PropTypes from 'prop-types';
import {Label} from 'react-bootstrap';
import moment from 'moment';
import 'moment-duration-format';

const TaskListHours = props => (
    <span className="task-list-hours">
      <Label bsStyle="info">
          {
              moment.duration(props.elapsedTime).format(
                  (props.elapsedTime > 60 * 60 * 1000) ? 'h[h] m[m]' : 'm[m] s[s]'
              )
          }
      </Label>
    </span>
);

TaskListHours.propTypes = {
    elapsedTime: PropTypes.number
};

export default TaskListHours;
