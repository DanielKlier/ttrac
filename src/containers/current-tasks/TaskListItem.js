import * as React from 'react';
import PropTypes from 'prop-types';
import TaskListHours from './TaskListHours';
import TaskListIssue from './TaskListIssue';
import {Button, Glyphicon} from 'react-bootstrap';
import TaskListTitle from './TaskListTitle';
import TimeCounter from './TimeCounter';

class TaskListItem extends React.PureComponent {
    // noinspection JSUnusedGlobalSymbols
    static PropTypes = {
        elapsedTime           : PropTypes.number,
        jiraIssue             : PropTypes.string,
        title                 : PropTypes.string,
        taskIsRunning         : PropTypes.bool,
        runningTask           : PropTypes.object,
        onTaskTitleChanged    : PropTypes.func.isRequired,
        onStartProgressClicked: PropTypes.func.isRequired,
        onStopProgressClicked : PropTypes.func.isRequired,
        onDeleteTaskClicked   : PropTypes.func.isRequired
    };

    render() {
        const props      = this.props;
        const classNames = [
            'list-group-item',
            'task-list-item'
        ];
        if (props.taskIsRunning) {
            classNames.push('running');
        }
        const className = classNames.join(' ');

        return (
            <li className={className}>
                <div className="task-list-item-badges">
                    {props.taskIsRunning &&
                    <TimeCounter startDate={props.runningTask.startDate}/>
                    }
                    {!props.taskIsRunning &&
                    <TaskListHours elapsedTime={props.elapsedTime}
                                   className="pull-left"/>
                    }
                    {props.jiraIssue &&
                    <TaskListIssue jiraIssue={props.jiraIssue}/>}
                </div>
                <div className="task-list-item-main">
                    <TaskListTitle title={props.title}
                                   onTextChanged={
                                       text => props.onTaskTitleChanged(text)
                                   }/>
                </div>
                <div className="task-list-item-buttons">
                    <Button onClick={() => props.onDeleteTaskClicked()}>
                        <Glyphicon glyph="trash"/>
                    </Button>
                    {props.taskIsRunning === false &&
                    <Button onClick={() => props.onStartProgressClicked()}>
                        <Glyphicon glyph="time"/>
                    </Button>
                    }
                    {props.taskIsRunning === true &&
                    <Button onClick={() => props.onStopProgressClicked()}>
                        <Glyphicon glyph="pause"/>
                    </Button>
                    }
                </div>
            </li>
        );
    }
}

export default TaskListItem;
