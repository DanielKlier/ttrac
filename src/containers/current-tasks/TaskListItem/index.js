import * as React from 'react';
import PropTypes from 'prop-types';
import {identity} from 'lodash';
import TaskListHours from './TaskListHours';
import TaskListIssue from './TaskListIssue';
import TaskListTitle from './TaskListTitle';
import TimeCounter from './TimeCounter';
import StartProgressButton from './StartProgressButton';
import StopProgressButton from './StopProgressButton';
import DeleteButton from '../../../components/DeleteButton';

class TaskListItem extends React.PureComponent {
    // noinspection JSUnusedGlobalSymbols
    static propTypes = {
        elapsedTime: PropTypes.number,
        jiraIssue: PropTypes.string,
        title: PropTypes.string,
        taskIsRunning: PropTypes.bool,
        runningTask: PropTypes.object,
        onTaskTitleChanged: PropTypes.func,
        onStartProgressClicked: PropTypes.func,
        onStopProgressClicked: PropTypes.func,
        onDeleteTaskClicked: PropTypes.func
    };

    // noinspection JSUnusedGlobalSymbols
    static defaultProps = {
        elapsedTime: 0,
        jiraIssue: '',
        title: '',
        taskIsRunning: false,
        runningTask: null,
        onTaskTitleChanged: identity,
        onStartProgressClicked: identity,
        onStopProgressClicked: identity,
        onDeleteTaskClicked: identity
    };

    render() {
        const props      = this.props;
        const classNames = [
            'list-group-item',
            'task-list-item'
        ];
        if (props.taskIsRunning) {
            classNames.push('running');
        }
        const className = classNames.join(' ');

        return (
            <li className={className}>
                <div className="task-list-item-badges">
                    {props.taskIsRunning &&
                    <TimeCounter startDate={props.runningTask.startDate}/>
                    }
                    {!props.taskIsRunning &&
                    <TaskListHours elapsedTime={props.elapsedTime}
                                   className="pull-left"/>
                    }
                    {props.jiraIssue &&
                    <TaskListIssue jiraIssue={props.jiraIssue}/>}
                </div>
                <div className="task-list-item-main">
                    <TaskListTitle title={props.title}
                                   onTextChanged={
                                       text => props.onTaskTitleChanged(text)
                                   }/>
                </div>
                <div className="task-list-item-buttons">
                    <DeleteButton onClick={() => props.onDeleteTaskClicked()}/>
                    {props.taskIsRunning === false &&
                    <StartProgressButton onClick={() => props.onStartProgressClicked()}/>
                    }
                    {props.taskIsRunning === true &&
                    <StopProgressButton onClick={() => props.onStopProgressClicked()}/>
                    }
                </div>
            </li>
        );
    }
}

export default TaskListItem;
