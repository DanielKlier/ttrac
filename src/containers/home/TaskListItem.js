import * as React from 'react';
import TaskListHours from './TaskListHours';
import TaskListIssue from './TaskListIssue';
import TaskListShortDescription from './TaskListShortDescription';
import {Button, Glyphicon} from 'react-bootstrap';

export default props => (
    <li className="list-group-item task-list-item">
        <div className="task-list-item-badges">
            <TaskListHours hours={props.hours} className="pull-left"/>
            {props.issue && <TaskListIssue issue={props.issue}/>}
        </div>
        <div className="task-list-item-main">
            <TaskListShortDescription description={props.description}/>
        </div>
        <div className="task-list-item-buttons">
            <Button>
                <Glyphicon glyph="time"/>
            </Button>
        </div>
    </li>
);