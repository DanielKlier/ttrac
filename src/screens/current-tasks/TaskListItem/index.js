import * as React from 'react';
import PropTypes from 'prop-types';
import {noop} from 'lodash';
import TaskListHours from './TaskListHours';
import TaskListIssue from './TaskListIssue';
import TaskListTitle from './TaskListTitle';
import TimeCounter from './TimeCounter';
import StartProgressButton from './StartProgressButton';
import StopProgressButton from './StopProgressButton';
import DeleteButton from '../../../components/DeleteButton';
import TaskListProject from './TaskListProject';

class TaskListItem extends React.PureComponent {
    // noinspection JSUnusedGlobalSymbols
    static propTypes = {
        elapsedTime: PropTypes.number,
        jiraIssue: PropTypes.string,
        title: PropTypes.string,
        taskIsRunning: PropTypes.bool,
        runningTask: PropTypes.object,
        onTaskTitleChanged: PropTypes.func,
        onStartProgressClicked: PropTypes.func,
        onStopProgressClicked: PropTypes.func,
        onDeleteTaskClicked: PropTypes.func,
        project: PropTypes.object,
        onProjectChanged: PropTypes.func
    };

    // noinspection JSUnusedGlobalSymbols
    static defaultProps = {
        elapsedTime: 0,
        jiraIssue: '',
        title: '',
        taskIsRunning: false,
        runningTask: null,
        onTaskTitleChanged: noop,
        onStartProgressClicked: noop,
        onStopProgressClicked: noop,
        onDeleteTaskClicked: noop,
        project: null,
        onProjectChanged: noop
    };

    render() {
        const props      = this.props;
        const classNames = [
            'list-group-item',
            'task-list-item'
        ];
        if (props.taskIsRunning) {
            classNames.push('running');
        }
        const className = classNames.join(' ');

        return (
            <li className={className}>
                <div className="task-list-item-badges">
                    {props.taskIsRunning &&
                    <TimeCounter startDate={props.runningTask.startDate}/>
                    }
                    {!props.taskIsRunning &&
                    <TaskListHours elapsedTime={props.elapsedTime}
                                   className="pull-left"/>
                    }
                    {props.jiraIssue &&
                    <TaskListIssue jiraIssue={props.jiraIssue}/>}
                </div>
                <div className="task-list-item-project">
                    <TaskListProject project={props.project}
                                     onProjectChanged={props.onProjectChanged}/>
                </div>
                <div className="task-list-item-main">
                    <TaskListTitle title={props.title}
                                   onTextChanged={
                                       text => props.onTaskTitleChanged(text)
                                   }/>
                </div>
                <div className="task-list-item-buttons">
                    <DeleteButton onClick={() => props.onDeleteTaskClicked()}/>
                    {props.taskIsRunning === false &&
                    <StartProgressButton onClick={() => props.onStartProgressClicked()}/>
                    }
                    {props.taskIsRunning === true &&
                    <StopProgressButton onClick={() => props.onStopProgressClicked()}/>
                    }
                </div>
            </li>
        );
    }
}

export default TaskListItem;
