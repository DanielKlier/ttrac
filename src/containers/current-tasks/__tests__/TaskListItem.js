import React from 'react';
import renderer from 'react-test-renderer';
import TaskListItem from '../TaskListItem';

Date.now = jest.fn(() => new Date('2017-03-09T19:33:00'));

test('TaskListItem renders correctly', () => {
    let component = renderer.create(
        <TaskListItem elapsedTime={3600000} jiraIssue="PRJ-123"
                      title="My Task"/>
    );
    let tree      = component.toJSON();

    expect(tree).toMatchSnapshot();

    component = renderer.create(
        <TaskListItem elapsedTime={3600000} jiraIssue="PRJ-123" title="My Task"
                      taskIsRunning={true}
                      runningTask={{
                          startDate: new Date('2017-03-09T18:00:00')
                      }}/>
    );
    tree      = component.toJSON();

    expect(tree).toMatchSnapshot();
});
