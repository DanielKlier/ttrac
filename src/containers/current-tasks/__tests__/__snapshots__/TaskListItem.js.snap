// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskListItem renders correctly 1`] = `
<li
  className="list-group-item task-list-item"
>
  <div
    className="task-list-item-badges"
  >
    <span
      className="task-list-hours"
    >
      <span
        className="label label-info"
      >
        60m 0s
      </span>
    </span>
    <span
      className="task-list-issue"
    >
      <a
        href="/"
      />
    </span>
  </div>
  <div
    className="task-list-item-main"
  >
    <h4
      className="task-list-description"
    >
      <span
        className=""
      >
        <span
          className="title"
          onClick={[Function]}
        >
          My Task
        </span>
      </span>
      <span
        className="hidden"
      >
        <input
          className="title-input"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Click to add a title"
          type="text"
          value="My Task"
        />
      </span>
    </h4>
  </div>
  <div
    className="task-list-item-buttons"
  >
    <button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span
        className="glyphicon glyphicon-trash"
      />
    </button>
  </div>
</li>
`;

exports[`TaskListItem renders correctly 2`] = `
<li
  className="list-group-item task-list-item running"
>
  <div
    className="task-list-item-badges"
  >
    <span
      className="task-list-hours"
    >
      <span
        className="label label-warning"
      >
        0s
      </span>
    </span>
    <span
      className="task-list-issue"
    >
      <a
        href="/"
      />
    </span>
  </div>
  <div
    className="task-list-item-main"
  >
    <h4
      className="task-list-description"
    >
      <span
        className=""
      >
        <span
          className="title"
          onClick={[Function]}
        >
          My Task
        </span>
      </span>
      <span
        className="hidden"
      >
        <input
          className="title-input"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Click to add a title"
          type="text"
          value="My Task"
        />
      </span>
    </h4>
  </div>
  <div
    className="task-list-item-buttons"
  >
    <button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span
        className="glyphicon glyphicon-trash"
      />
    </button>
    <button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span
        className="glyphicon glyphicon-pause"
      />
    </button>
  </div>
</li>
`;
